ForkMe-Mobile/app/src/main/java/dlei/forkme/gui/activities/github/UserRepositoriesViewActivity.java

```diff
@@ -0,0 +1,291 @@
+                            mUserStarIconImage.getContext(), R.color.primary_light));
+                    getUserRepositories(userLogin);
+                } else {
+                    Log.w("UserReposViewActivity: ", String.format(
+                            "getGithubUser: Error: Status code: %d, successful: %s," + "headers: %s",
+                            response.code(), response.isSuccessful(), response.headers())
+                    );
+                }
+
+            }
+
+            @Override
+            public void onFailure(Call<User> call, Throwable t) {
+                // Failure to connect to endpoint.
+                Log.i("UserActivity: ", "getGithubUser: Failed: " + t.getMessage());
+                NetworkAsyncCheck n = NetworkHelper.checkNetworkConnection(mAvatarIconImage);
+                if (n != null) {
+                    n.execute();
+                }
+
+            }
+        });
+    }
+
+    /**
+     * Get a user's GitHub repositories (restricted to public if the user is not the authenticated user.
+     * @param userLogin GitHub login for the user.
+     */
+    public void getUserRepositories(final String userLogin) {
+        // Goes into the network level of OkHttp.
+        OkHttpClient.Builder okHttpBuilder = new OkHttpClient.Builder();
+        okHttpBuilder.addInterceptor(new Interceptor() {
+            @Override
+            public okhttp3.Response intercept(Chain chain) throws IOException {
+                // Manipulate request to add headers.
+                // Can't mutate the request but can make a new one.
+                Request request = chain.request();
+                Request.Builder newRequest = request.newBuilder()
+                        // Add in user access token.
+                        .addHeader("Authorization", "token " + AppSettings.sOAuthToken);
+                // Pass on our request to execute.
+                return chain.proceed(newRequest.build());
+            }
+        });
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .client(okHttpBuilder.build())
+                .baseUrl(BaseUrls.githubApi)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+
+        GithubApi endpoint = retrofit.create(GithubApi.class);
+        endpoint.getMyRepositories(userLogin).enqueue(new Callback<List<Repository>>() {
+            @Override
+            public void onResponse(Call<List<Repository>> call, Response<List<Repository>> response) {
+
+                if (response.code() == 200 && response.isSuccessful()) {
+                    ArrayList<Repository> repositories = (ArrayList<Repository>) response.body();
+
+                    // Add starred repositories to array.
+                    for (Repository r: repositories) {
+                        mRepositories.add(r);
+                    }
+                    // Notify data set changed.
+                    mAdapterRepositories.notifyDataSetChanged();
```

Comment:
I suggest using notifyItemRangeInserted(...): https://developer.android.com/reference/android/support/v7/widget/RecyclerView.Adapter.html#notifyItemRangeInserted(int, int)
It typically provides better performance and allows you to animate the insertions of items.